llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -303,6 +309,13 @@ class Value {
 #endif
 };
 
+/// Helper Attorney-Client class that gives access to the underlying IR.
```

Comment:
Could the `ValueAttorney` be in the cpp file? It is kind of a public loop hole.

Reply:
It will only give access to the LLVM Value to its friends, so I don't see how it is a loop hole.

Reply:
can we add `friend class User;` in `Value` instead?

Reply:
Yes we can, and in this case it makes sense to use a friend because the User class is expected to be tightly coupled to the Value class. But for classes that are not as closely related to the IR, like the transaction tracker, it is nicer to use a client-attorney class to restrict access to just the LLVM Value and nothing else. Anyway, I will remove it for now and add it in later.